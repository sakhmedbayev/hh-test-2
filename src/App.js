/* eslint-disable jsx-a11y/no-onchange */
import React, { useReducer } from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";

const sharedRoomLabelStyles = css`
  padding: 3px;
  font-weight: bold;
  font-size: 12px;
`;

// App RoomStatus component

const StyledRoomStatusLabel = styled.label`
  ${sharedRoomLabelStyles};
  background-color: ${props => (props.checked ? "#e7e7e7" : "#DBDBE3")};
`;

StyledRoomStatusLabel.propTypes = {
  checked: PropTypes.bool.isRequired
};

export const RoomCheckInStatusBar = ({ roomStatus, onRoomStatusChange }) => {
  return (
    <StyledRoomStatusLabel
      htmlFor={`roomStatus-${roomStatus.roomNumber}`}
      aria-label={`Room ${roomStatus.roomNumber}`}
      checked={roomStatus.checked}
      data-testid={`room-${roomStatus.roomNumber}-status-label`}
    >
      <input
        name="room-status"
        type="checkbox"
        id={`roomStatus-${roomStatus.roomNumber}`}
        checked={roomStatus.checked}
        onChange={e => {
          onRoomStatusChange({ ...roomStatus, checked: e.target.checked });
        }}
        data-testid={`room-${roomStatus.roomNumber}-checkbox`}
      />
      <span style={{ marginLeft: "5px" }}>Room {roomStatus.roomNumber}</span>
    </StyledRoomStatusLabel>
  );
};

RoomCheckInStatusBar.propTypes = {
  roomStatus: PropTypes.shape({
    roomNumber: PropTypes.number.isRequired,
    checked: PropTypes.bool.isRequired,
    numAdults: PropTypes.string.isRequired,
    numChildren: PropTypes.string.isRequired
  }),
  onRoomStatusChange: PropTypes.func.isRequired
};

// GuestNumber component

const StyledDiv2 = styled.div`
  display: flex;
  flex-direction: column;
  width: 50px;
`;

const StyledGuestLabel = styled.label`
  font-size: 12px;
  margin-bottom: 5px;
`;

export const GuestNumberDropdown = ({
  labelPrefix,
  labelSuffix,
  options,
  roomStatus,
  onRoomStatusChange,
  type
}) => {
  return (
    <StyledDiv2 role={`room-dropdown-${labelPrefix}-${labelSuffix}`}>
      <StyledGuestLabel
        htmlFor={`room-dropdown-${labelPrefix}-${labelSuffix}`}
        area-label={`${labelPrefix} ${labelSuffix}`}
      >
        {labelPrefix} <br /> {labelSuffix}
      </StyledGuestLabel>
      <select
        name="number-of-guest-status"
        id={`room-dropdown-${labelPrefix}-${labelSuffix}`}
        value={roomStatus[type]}
        onChange={e => {
          onRoomStatusChange({ ...roomStatus, [type]: e.target.value });
        }}
        disabled={roomStatus.roomNumber !== 1 && !roomStatus.checked}
        data-testid={`room-dropdown-${roomStatus.roomNumber}-${labelPrefix}`}
      >
        {options.map(option => (
          <option
            key={option}
            value={option}
            data-testid={`option-${labelPrefix}-${roomStatus.roomNumber}-${option}`}
          >
            {option}
          </option>
        ))}
      </select>
    </StyledDiv2>
  );
};

GuestNumberDropdown.propTypes = {
  labelPrefix: PropTypes.string.isRequired,
  labelSuffix: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string),
  roomStatus: PropTypes.shape({
    roomNumber: PropTypes.number.isRequired,
    checked: PropTypes.bool.isRequired,
    numAdults: PropTypes.string.isRequired,
    numChildren: PropTypes.string.isRequired
  }),
  type: PropTypes.string.isRequired,
  onRoomStatusChange: PropTypes.func.isRequired
};

// RoomWidget component

const StyledHorizontalDiv = styled.div`
  display: flex;
  flex-direction: row;
  padding: 10px;
  div:first-child {
    margin-right: 10px;
  }
`;

const StyledSectionForRoomWidget = styled.div`
  background-color: ${props => (props.checked ? "" : "#DBDBE3")};
  border-radius: 6px;
  border: ${props =>
    props.checked ? "3px solid #e7e7e7" : "3px solid #CBD0DB"};
  display: flex;
  flex-direction: column;
  height: 100%;
  margin-bottom: 10px;
  margin-right: 10px;
  min-width: 150px;
`;

StyledSectionForRoomWidget.propTypes = {
  checked: PropTypes.bool.isRequired
};

const StyledRoomLabelSpan = styled.span`
  ${sharedRoomLabelStyles};
  padding: 4px 0px 3px 5px;
  background-color: #e7e7e7;
`;

export const RoomWidget = ({ roomStatus, onRoomStatusChange }) => {
  return (
    <StyledSectionForRoomWidget
      role={`check-in room number ${roomStatus.roomNumber}`}
      aria-disabled={
        roomStatus.roomNumber === "1" ? false : !roomStatus.checked
      }
      checked={roomStatus.roomNumber === "1" ? true : roomStatus.checked}
      data-testid={`room-${roomStatus.roomNumber}-widget-container`}
    >
      {roomStatus.roomNumber === 1 ? (
        <StyledRoomLabelSpan data-testid="room-1-status-label">
          Room 1
        </StyledRoomLabelSpan>
      ) : (
        <RoomCheckInStatusBar
          roomStatus={roomStatus}
          onRoomStatusChange={onRoomStatusChange}
        />
      )}
      <StyledHorizontalDiv>
        <GuestNumberDropdown
          labelPrefix="Adults"
          labelSuffix="(18+)"
          options={["1", "2"]}
          roomStatus={roomStatus}
          onRoomStatusChange={onRoomStatusChange}
          type="numAdults"
        />
        <GuestNumberDropdown
          labelPrefix="Children"
          labelSuffix="(0-17)"
          options={["0", "1", "2"]}
          roomStatus={roomStatus}
          onRoomStatusChange={onRoomStatusChange}
          type="numChildren"
        />
      </StyledHorizontalDiv>
    </StyledSectionForRoomWidget>
  );
};

RoomWidget.propTypes = {
  roomStatus: PropTypes.shape({
    roomNumber: PropTypes.number.isRequired,
    checked: PropTypes.bool.isRequired,
    numAdults: PropTypes.string.isRequired,
    numChildren: PropTypes.string.isRequired
  }),
  onRoomStatusChange: PropTypes.func.isRequired
};

// App component

const StyledMain = styled.main`
  padding: 20px;
`;

const StyledForm = styled.form``;

const StyledRoomsWidgetDiv = styled.div`
  align-content: center;
  align-items: flex-start;
  display: flex;
  flex-grow: 1;
  flex-wrap: wrap;
  justify-content: end;
  @media (max-width: 600px) {
    .flex-container > div {
      width: 100%;
    }
  }
`;

export const initialState = {
  roomStatuses: [
    {
      roomNumber: 1,
      checked: true,
      numAdults: "1",
      numChildren: "0"
    },
    {
      roomNumber: 2,
      checked: false,
      numAdults: "1",
      numChildren: "0"
    },
    {
      roomNumber: 3,
      checked: false,
      numAdults: "1",
      numChildren: "0"
    },
    {
      roomNumber: 4,
      checked: false,
      numAdults: "1",
      numChildren: "0"
    }
  ]
};

const RESET = "RESET";
const CHANGE_ROOM_STATUSES = "CHANGE_ROOM_STATUSES";

const reducer = (state, action) => {
  switch (action.type) {
    case "GET_ROOM_STATUSES": {
      return { ...state, roomStatuses: action.roomStatuses };
    }
    case CHANGE_ROOM_STATUSES: {
      const updatedRoomStatuses = state.roomStatuses.map((prevRoom, index) => {
        if (
          prevRoom.roomNumber < action.payload.roomNumber &&
          action.payload.checked
        ) {
          return { ...prevRoom, checked: true };
        } else if (
          prevRoom.roomNumber >= action.payload.roomNumber &&
          !action.payload.checked
        ) {
          return {
            roomNumber: prevRoom.roomNumber,
            checked: false,
            numAdults: "1",
            numChildren: "0"
          };
        } else if (index !== action.payload.roomNumber - 1) {
          return { ...prevRoom };
        }
        return {
          ...prevRoom,
          ...action.payload
        };
      });

      return {
        ...state,
        roomStatuses: updatedRoomStatuses
      };
    }
    case RESET: {
      return init();
    }
    default:
      return state;
  }
};

// 1. include prop types

function init() {
  const data = JSON.parse(localStorage.getItem("roomStatuses"));
  const initialData = initialState.roomStatuses.map(room => {
    if (data && data[room.roomNumber - 1]) {
      return data[room.roomNumber - 1];
    }
    return room;
  });
  return { roomStatuses: data ? initialData : initialState.roomStatuses };
}

const App = () => {
  const [{ roomStatuses }, dispatch] = useReducer(reducer, initialState, init);

  return (
    <StyledMain>
      {/* This should be included for better accessibility */}
      <h1 data-testid="h1tag">Hilton Hotel rooms online check-in</h1>
      <StyledForm
        role="check-in form"
        onSubmit={e => {
          e.preventDefault();

          const allCheckedInRooms = roomStatuses.filter(
            room => room.checked === true
          );
          localStorage.setItem(
            "roomStatuses",
            JSON.stringify(allCheckedInRooms)
          );
        }}
      >
        <StyledRoomsWidgetDiv role="check-in room form">
          {roomStatuses.map(room => (
            <RoomWidget
              key={room.roomNumber}
              roomStatus={room}
              onRoomStatusChange={values => {
                dispatch({ type: CHANGE_ROOM_STATUSES, payload: values });
              }}
            />
          ))}
        </StyledRoomsWidgetDiv>
        <button type="submit">Submit</button>
        <button
          onClick={() => {
            dispatch({ type: RESET });
          }}
        >
          Reset
        </button>
      </StyledForm>
    </StyledMain>
  );
};

export default App;
