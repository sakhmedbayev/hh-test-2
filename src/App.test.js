import { LocalStorageMock } from "@react-mock/localstorage";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, fireEvent, render, waitForElement } from "@testing-library/react";
import React from "react";
import App, { initialState } from "./App";

afterEach(cleanup);

const renderComponent = data =>
  render(
    <LocalStorageMock items={data}>
      <App />
    </LocalStorageMock>
  );

const allTogglableRooms = initialState.roomStatuses.slice(1);

describe("elements render properly", () => {
  it("renders according to the snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders togglable rooms with their default/disabled properties", () => {
    const { getByTestId } = render(<App />);

    allTogglableRooms.forEach(room => {
      const roomCheckboxTestId = `room-${room.roomNumber}-checkbox`;
      const roomWidgetTestId = `room-${room.roomNumber}-widget-container`;
      const guestNumDropdownAdultsTestId = `room-dropdown-${room.roomNumber}-Adults`;
      const guestNumDropdownChildrenTestId = `room-dropdown-${room.roomNumber}-Children`;

      expect(getByTestId(roomCheckboxTestId)).toHaveProperty("checked", false);
      expect(getByTestId(guestNumDropdownAdultsTestId)).toHaveProperty(
        "disabled",
        true
      );
      expect(getByTestId(guestNumDropdownChildrenTestId)).toHaveProperty(
        "disabled",
        true
      );
      expect(getByTestId(roomWidgetTestId)).toHaveStyle(
        "background-color: #DBDBE3"
      );
    });
  });
});

describe("Room widget", () => {
  it("changes checkbox status for room 2 correctly", async () => {
    const component = render(<App />);

    fireEvent.click(component.getByTestId(`room-2-status-label`));

    const updatedComponent = await waitForElement(() =>
      component.getByTestId(`room-2-status-label`)
    );

    expect(updatedComponent).toHaveProperty("checked", true);
  });
  it("changes checkbox status for room 3 correctly", async () => {
    const component = render(<App />);

    fireEvent.click(component.getByTestId(`room-3-status-label`));

    const updatedComponent = await waitForElement(() =>
      component.getByTestId(`room-3-status-label`)
    );

    expect(updatedComponent).toHaveProperty("checked", true);
  });
  it("changes checkbox status for room 4 correctly", async () => {
    const component = render(<App />);

    fireEvent.click(component.getByTestId(`room-4-status-label`));

    const updatedComponent = await waitForElement(() =>
      component.getByTestId(`room-4-status-label`)
    );

    expect(updatedComponent).toHaveProperty("checked", true);
  });
  it("sets proper value for room 1 adults dropdown correctly", () => {
    const component = render(<App />);

    const foundSelectBeforeEvent = component.getByTestId(
      `room-dropdown-1-Adults`
    );

    fireEvent.change(foundSelectBeforeEvent, { target: { value: "2" } });

    expect(foundSelectBeforeEvent.value).toBe("2");
  });
  it("sets proper value for room 1 children dropdown correctly", () => {
    const component = render(<App />);

    const foundSelectBeforeEvent = component.getByTestId(
      `room-dropdown-1-Children`
    );

    fireEvent.change(foundSelectBeforeEvent, { target: { value: "2" } });

    expect(foundSelectBeforeEvent.value).toBe("2");
  });
  it("sets proper value for room 2 adults dropdown correctly", () => {
    const component = render(<App />);
    fireEvent.click(component.getByTestId(`room-2-status-label`));
    const foundSelectBeforeEvent = component.getByTestId(
      `room-dropdown-2-Adults`
    );
    fireEvent.change(foundSelectBeforeEvent, { target: { value: "2" } });
    expect(foundSelectBeforeEvent.value).toBe("2");
  });
  it("sets proper value for room 1 children dropdown correctly", () => {
    const component = render(<App />);
    fireEvent.click(component.getByTestId(`room-2-status-label`));
    const foundSelectBeforeEvent = component.getByTestId(
      `room-dropdown-2-Children`
    );
    fireEvent.change(foundSelectBeforeEvent, { target: { value: "2" } });
    expect(foundSelectBeforeEvent.value).toBe("2");
  });
});

describe("a room's checked status has proper effect on other rooms' checked statuses", () => {
  it("sets proper checked status on the current and other rooms", () => {
    const component = render(<App />);

    fireEvent.click(component.getByTestId("room-4-checkbox"));

    expect(component.getByTestId("room-2-checkbox")).toHaveProperty(
      "checked",
      true
    );
    expect(component.getByTestId("room-3-checkbox")).toHaveProperty(
      "checked",
      true
    );
    expect(component.getByTestId("room-4-checkbox")).toHaveProperty(
      "checked",
      true
    );

    fireEvent.click(component.getByTestId("room-2-checkbox"));

    expect(component.getByTestId("room-2-checkbox")).toHaveProperty(
      "checked",
      false
    );
    expect(component.getByTestId("room-3-checkbox")).toHaveProperty(
      "checked",
      false
    );
    expect(component.getByTestId("room-4-checkbox")).toHaveProperty(
      "checked",
      false
    );
  });
});

describe("the form initialized correctly by pulling data from LocalStorage", () => {
  it("pulls data from LocalStorage and sets proper statuses on rooms' checkboxes", () => {
    const dataForLocalStorage = [
      { roomNumber: 1, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 2, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 3, checked: false, numAdults: "1", numChildren: "0" },
      { roomNumber: 4, checked: false, numAdults: "1", numChildren: "0" }
    ];

    const component = renderComponent({
      roomStatuses: JSON.stringify(dataForLocalStorage)
    });

    expect(component.getByTestId("room-2-checkbox")).toHaveProperty(
      "checked",
      true
    );
    expect(component.getByTestId("room-3-checkbox")).toHaveProperty(
      "checked",
      false
    );
    expect(component.getByTestId("room-4-checkbox")).toHaveProperty(
      "checked",
      false
    );
  });
});

describe("form submissions", () => {
  it("updates LocalStorage cache when only room 1 is submitted", () => {
    const dataForLocalStorage = [
      { roomNumber: 1, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 2, checked: false, numAdults: "1", numChildren: "0" },
      { roomNumber: 3, checked: false, numAdults: "1", numChildren: "0" },
      { roomNumber: 4, checked: false, numAdults: "1", numChildren: "0" }
    ];

    const component = renderComponent({
      roomStatuses: JSON.stringify(dataForLocalStorage)
    });

    fireEvent.click(component.getByText(/Submit/i));

    // LocalStorage should store only room 1 data
    expect(JSON.parse(localStorage.getItem("roomStatuses"))).toStrictEqual(
      dataForLocalStorage.slice(0, 1)
    );
  });
  it("updates LocalStorage cache when room 1 and 2 are submitted", () => {
    const dataForLocalStorage = [
      { roomNumber: 1, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 2, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 3, checked: false, numAdults: "1", numChildren: "0" },
      { roomNumber: 4, checked: false, numAdults: "1", numChildren: "0" }
    ];

    const component = renderComponent({
      roomStatuses: JSON.stringify(dataForLocalStorage)
    });

    fireEvent.click(component.getByText(/Submit/i));

    // LocalStorage should store data for room 1 and 2
    expect(JSON.parse(localStorage.getItem("roomStatuses"))).toStrictEqual(
      dataForLocalStorage.slice(0, 2)
    );
  });
  it("updates LocalStorage when room 1 and 2 are checked initially but room 3 and 4 are set true by a user and submitted with their default values", () => {
    const dataForLocalStorage = [
      { roomNumber: 1, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 2, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 3, checked: false, numAdults: "1", numChildren: "0" },
      { roomNumber: 4, checked: false, numAdults: "1", numChildren: "0" }
    ];

    const component = renderComponent({
      roomStatuses: JSON.stringify(dataForLocalStorage)
    });

    fireEvent.click(component.getByTestId("room-3-checkbox"));
    fireEvent.click(component.getByTestId("room-4-checkbox"));
    fireEvent.click(component.getByText(/Submit/i));

    expect(JSON.parse(localStorage.getItem("roomStatuses"))).toStrictEqual([
      ...dataForLocalStorage.slice(0, 2),
      { roomNumber: 3, checked: true, numAdults: "1", numChildren: "0" },
      { roomNumber: 4, checked: true, numAdults: "1", numChildren: "0" }
    ]);
  });
});
