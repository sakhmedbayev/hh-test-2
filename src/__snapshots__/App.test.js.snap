// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements render properly renders according to the snapshot 1`] = `
<DocumentFragment>
  <main
    class="sc-bZQynM hvLcDR"
  >
    <h1
      data-testid="h1tag"
    >
      Hilton Hotel rooms online check-in
    </h1>
    <form
      class="sc-gzVnrw ibZhkv"
      role="check-in form"
    >
      <div
        class="sc-htoDjs fObrRs"
        role="check-in room form"
      >
        <div
          aria-disabled="false"
          class="sc-ifAKCX fljDJW"
          data-testid="room-1-widget-container"
          role="check-in room number 1"
        >
          <span
            class="sc-EHOje hgpcZa"
            data-testid="room-1-status-label"
          >
            Room 1
          </span>
          <div
            class="sc-bxivhb emBooZ"
          >
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Adults-(18+)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Adults-(18+)"
              >
                Adults 
                <br />
                 (18+)
              </label>
              <select
                data-testid="room-dropdown-1-Adults"
                id="room-dropdown-Adults-(18+)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Adults-1-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Adults-1-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Children-(0-17)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Children-(0-17)"
              >
                Children 
                <br />
                 (0-17)
              </label>
              <select
                data-testid="room-dropdown-1-Children"
                id="room-dropdown-Children-(0-17)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Children-1-0"
                  value="0"
                >
                  0
                </option>
                <option
                  data-testid="option-Children-1-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Children-1-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          aria-disabled="true"
          class="sc-ifAKCX dFuSiM"
          data-testid="room-2-widget-container"
          role="check-in room number 2"
        >
          <label
            aria-label="Room 2"
            class="sc-bdVaJa iCGeOX"
            data-testid="room-2-status-label"
            for="roomStatus-2"
          >
            <input
              data-testid="room-2-checkbox"
              id="roomStatus-2"
              name="room-status"
              type="checkbox"
            />
            <span
              style="margin-left: 5px;"
            >
              Room 2
            </span>
          </label>
          <div
            class="sc-bxivhb emBooZ"
          >
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Adults-(18+)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Adults-(18+)"
              >
                Adults 
                <br />
                 (18+)
              </label>
              <select
                data-testid="room-dropdown-2-Adults"
                disabled=""
                id="room-dropdown-Adults-(18+)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Adults-2-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Adults-2-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Children-(0-17)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Children-(0-17)"
              >
                Children 
                <br />
                 (0-17)
              </label>
              <select
                data-testid="room-dropdown-2-Children"
                disabled=""
                id="room-dropdown-Children-(0-17)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Children-2-0"
                  value="0"
                >
                  0
                </option>
                <option
                  data-testid="option-Children-2-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Children-2-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          aria-disabled="true"
          class="sc-ifAKCX dFuSiM"
          data-testid="room-3-widget-container"
          role="check-in room number 3"
        >
          <label
            aria-label="Room 3"
            class="sc-bdVaJa iCGeOX"
            data-testid="room-3-status-label"
            for="roomStatus-3"
          >
            <input
              data-testid="room-3-checkbox"
              id="roomStatus-3"
              name="room-status"
              type="checkbox"
            />
            <span
              style="margin-left: 5px;"
            >
              Room 3
            </span>
          </label>
          <div
            class="sc-bxivhb emBooZ"
          >
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Adults-(18+)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Adults-(18+)"
              >
                Adults 
                <br />
                 (18+)
              </label>
              <select
                data-testid="room-dropdown-3-Adults"
                disabled=""
                id="room-dropdown-Adults-(18+)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Adults-3-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Adults-3-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Children-(0-17)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Children-(0-17)"
              >
                Children 
                <br />
                 (0-17)
              </label>
              <select
                data-testid="room-dropdown-3-Children"
                disabled=""
                id="room-dropdown-Children-(0-17)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Children-3-0"
                  value="0"
                >
                  0
                </option>
                <option
                  data-testid="option-Children-3-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Children-3-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          aria-disabled="true"
          class="sc-ifAKCX dFuSiM"
          data-testid="room-4-widget-container"
          role="check-in room number 4"
        >
          <label
            aria-label="Room 4"
            class="sc-bdVaJa iCGeOX"
            data-testid="room-4-status-label"
            for="roomStatus-4"
          >
            <input
              data-testid="room-4-checkbox"
              id="roomStatus-4"
              name="room-status"
              type="checkbox"
            />
            <span
              style="margin-left: 5px;"
            >
              Room 4
            </span>
          </label>
          <div
            class="sc-bxivhb emBooZ"
          >
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Adults-(18+)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Adults-(18+)"
              >
                Adults 
                <br />
                 (18+)
              </label>
              <select
                data-testid="room-dropdown-4-Adults"
                disabled=""
                id="room-dropdown-Adults-(18+)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Adults-4-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Adults-4-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
            <div
              class="sc-bwzfXH idzafn"
              role="room-dropdown-Children-(0-17)"
            >
              <label
                class="sc-htpNat kOrLIv"
                for="room-dropdown-Children-(0-17)"
              >
                Children 
                <br />
                 (0-17)
              </label>
              <select
                data-testid="room-dropdown-4-Children"
                disabled=""
                id="room-dropdown-Children-(0-17)"
                name="number-of-guest-status"
              >
                <option
                  data-testid="option-Children-4-0"
                  value="0"
                >
                  0
                </option>
                <option
                  data-testid="option-Children-4-1"
                  value="1"
                >
                  1
                </option>
                <option
                  data-testid="option-Children-4-2"
                  value="2"
                >
                  2
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
      <button>
        Reset
      </button>
    </form>
  </main>
</DocumentFragment>
`;
